src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -1362,6 +1362,27 @@ namespace {
 
     ASSERT_TRUE(!check_vector_eq(activeColumns, activeColumnsGlobal));
     ASSERT_TRUE(check_vector_eq(activeColumns, activeColumnsLocal));
+
+    // Assert that only columns with overlap are activated
+    numColumns = 16;
+    density = 0.5;
+    inhibitionRadius = 2;
+    Real overlapsArray2[16] = {1, 1, 1, 0, 1, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0};
+
+    sp.setGlobalInhibition(false);
+    sp.setInhibitionRadius(inhibitionRadius);
+
+    overlapsReal.assign(&overlapsArray2[0], &overlapsArray2[numColumns]);
+    sp.inhibitColumnsLocal_(overlapsReal, density, activeColumnsLocal);
+
+    for (auto & activeColumn : activeColumnsLocal) {
+      cout << "\n================================================";
```

Comment:
Instead of dumping to `cout`, you can dump an error message to `ASSERT_TRUE` which avoids a bunch of extraneous output during successful test runs.